membership-common/src/main/scala/com/gu/memsub/subsv2/Subscription.scala

```diff
@@ -13,22 +13,18 @@ import scalaz.{NonEmptyList, Validation, \/}
 
 case class Subscription(
     id: memsub.Subscription.Id,
-    name: memsub.Subscription.Name,
+    subscriptionNumber: memsub.Subscription.Name,
     accountId: memsub.Subscription.AccountId,
-    startDate: LocalDate,
-    acceptanceDate: LocalDate,
-    termStartDate: LocalDate,
+    contractEffectiveDate: LocalDate,
+    customerAcceptanceDate: LocalDate,
     termEndDate: LocalDate,
-    casActivationDate: Option[DateTime],
-    promoCode: Option[PromoCode],
     isCancelled: Boolean,
     ratePlans: List[RatePlan],
     readerType: ReaderType,
-    gifteeIdentityId: Option[String],
     autoRenew: Boolean,
```

Comment:
these are renamed to pretty much match the zuora model.  I found that the removed fields were not actually used so I removed them for clarity.  There's no reason not to add them back later if we want them.